client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/index.tsx

```diff
@@ -132,20 +140,20 @@ const SiteCreationStep: Step = function SiteCreationStep( { navigation, flow, da
 			true,
 			username,
 			domainCartItem,
-			data?.sourceSlug as string
+			sourceSlug
```

Comment:
Do we need to update the `sourceSlug` to consider undefined?

<img width="349" alt="CleanShot 2023-04-25 at 13 57 31@2x" src="https://user-images.githubusercontent.com/10482592/234362110-bb629e92-5397-4bdc-9354-d87c707e545b.png">

Reply:
The question mark after `sourceSlug` counts as `undefined` so it is already supported.